### PR TITLE
add monitoring plugin

### PR DESCRIPTION
**Pull request by timogoebel**

### CODE REVIEW

**Comment by ekohl** [comment]
What confuses me a bit is that there's a monitoring plugin and an icinga2 monitoring plugin. Can you use the icinga2 plugin without the base monitoring plugin? It only drops a config file so maybe it should be part of the monitoring class as optional parameters?

File: manifests/plugin/monitoring/icinga2/params.pp [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class foreman_proxy::plugin::monitoring::icinga2::params {
+  $enabled = true
+  $server = 'icinga2.localdomain'
```
Comment (ekohl) [inline manifests/plugin/monitoring/icinga2/params.pp]: Maybe default to the fqdn? That's more likely to work than this I think.

**Comment by timogoebel** [comment]
> Can you use the icinga2 plugin without the base monitoring plugin?

No, you cannot. However there will be other monitoring providers in the future. Right now the code for smart_proxy_monitoring and smart_proxy_monitoring_icinga2 is in the same gem for easier development. But as soon as there is another provider, we'll extract it out of there.

**Comment by ekohl** [comment]
Should the icinga2 class include the monitoring class then? I'm afraid users will include icinga2 and expect it to just work.

**Comment by timogoebel** [comment]
> Should the icinga2 class include the monitoring class then?

Yes, we could do that. That makes sense.

**Comment by timogoebel** [comment]
@ekohl : Updated the code to address your suggestions. Thanks for taking a look!

File: templates/plugin/monitoring_icinga2.yml.erb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+---
+:enabled: <%= scope.lookupvar('::foreman_proxy::plugin::monitoring::icinga2::enabled') %>
+:server: "<%= scope.lookupvar('::foreman_proxy::plugin::monitoring::icinga2::server') %>"
+:api_cacert: "<%= scope.lookupvar('::foreman_proxy::plugin::monitoring::icinga2::api_cacert') %>"
+:api_user: "<%= scope.lookupvar('::foreman_proxy::plugin::monitoring::icinga2::api_user') %>"
+<% api_password = scope.lookupvar('::foreman_proxy::plugin::monitoring::icinga2::api_password') -%>
+<% if api_password.nil? || api_password == :undef %>
```
Comment (ekohl) [inline templates/plugin/monitoring_icinga2.yml.erb]: Maybe also check if it's non-empty?

**Comment by ekohl** [comment]
:+1: looks good.

File: manifests/plugin/monitoring/icinga2.pp [hunk 3]
```diff
@@ -0,0 +1,49 @@
+# = Foreman Proxy Icinga 2 Monitoring plugin
+#
+# This class installs the Icinga 2 Monitoring plugin for Foreman proxy
+#
+# === Parameters:
+#
+# $enabled::               Enable this plugin.
```
Comment (domcleal) [inline manifests/plugin/monitoring/icinga2.pp]: Move `enabled` into an Advanced section like with the main monitoring class, so it's hidden by default in the installer.

File: manifests/plugin/monitoring/icinga2.pp [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+# $api_cacert::            Path to Icinga2 server CA certificate file.
+#
+# $api_user::              Icinga2 API username.
+#
+# $api_usercert::          Path to Icinga2 user certificate file.
+#
+# $api_userkey::           Path to Icinga2 user key file.
+#
+# $api_password::          Icinga2 API password. If set to undef (default) API
+#                          connection is made via certificate and key.
```
Comment (domcleal) [inline manifests/plugin/monitoring/icinga2.pp]: add `type:password`

File: manifests/plugin/monitoring/params.pp [hunk 5]
```diff
@@ -0,0 +1,10 @@
+# Default parameters for the Monitoring smart proxy plugin
+class foreman_proxy::plugin::monitoring::params {
+  include ::foreman_proxy::params
```
Comment (domcleal) [inline manifests/plugin/monitoring/params.pp]: unused

File: manifests/plugin/monitoring/icinga2.pp (hunk 3 above)
Comment (timogoebel) [inline manifests/plugin/monitoring/icinga2.pp]: done.

File: manifests/plugin/monitoring/icinga2.pp (hunk 4 above)
Comment (timogoebel) [inline manifests/plugin/monitoring/icinga2.pp]: done.

File: templates/plugin/monitoring_icinga2.yml.erb [hunk 6]
```diff
@@ -0,0 +1,13 @@
+---
+:enabled: <%= scope.lookupvar('::foreman_proxy::plugin::monitoring::icinga2::enabled') %>
+:server: "<%= scope.lookupvar('::foreman_proxy::plugin::monitoring::icinga2::server') %>"
+:api_cacert: "<%= scope.lookupvar('::foreman_proxy::plugin::monitoring::icinga2::api_cacert') %>"
+:api_user: "<%= scope.lookupvar('::foreman_proxy::plugin::monitoring::icinga2::api_user') %>"
+<% api_password = scope.lookupvar('::foreman_proxy::plugin::monitoring::icinga2::api_password') -%>
+<% if (api_password.nil? && !api_password.empty?) || api_password == :undef %>
```
Comment (domcleal) [inline templates/plugin/monitoring_icinga2.yml.erb]: This is raising an error in tests when api_password is nil. I'd suggest: `if api_password.nil? || api_password == :undef || api_password.empty?`

Comment (timogoebel) [inline templates/plugin/monitoring_icinga2.yml.erb]: Followed your suggestion. Thanks, @domcleal .

**Comment by timogoebel** [comment]
Tests are fine now. Are we ready to 🚢 this?

**Comment by ekohl** [comment]
Thanks!

**Pull request closed without merging**